src/dos/drive_virtual.cpp

```diff
@@ -386,20 +392,12 @@ bool Virtual_Drive::FileOpen(DOS_File * * file,char * name,uint32_t flags) {
 		return false;
 	}
 /* Scan through the internal list of files */
-	auto cur_file = first_file;
-	while (cur_file) {
-		unsigned int onpos = cur_file->onpos;
-		if (strcasecmp(name, (std::string(onpos ? vfilenames[onpos].shortname +
-		                                                  std::string(1, '\\')
-		                                        : "") +
-		                      cur_file->name)
-		                             .c_str()) == 0) {
-			/* We have a match */
-			*file = new Virtual_File(cur_file->data, cur_file->size);
-			(*file)->flags = flags;
-			return true;
-		}
-		cur_file = cur_file->next;
+	auto found = find_file_by_name(name);
+	if (found) {
```

Comment:
Suggest calling the `find_file_by_name()` helper `vfile_name_exists(...)`.

And then collapsing out the `found` teporary in all its uses:

```c++
if (vfile_name_exists(name)) {
    ...
}
```

Reply:
some of the usages need data from the resulting vfile like size - i will add a single line wrapper of the just-exists test